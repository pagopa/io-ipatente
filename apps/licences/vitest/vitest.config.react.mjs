/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",

        // BE
        "src/app/*",
        "src/config/*",
        "src/lib/bff/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.{tsx,ts}"],
      reporter: ["text", "html"],
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
