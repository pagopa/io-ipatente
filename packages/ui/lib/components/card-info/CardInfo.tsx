import { Divider, List, ListItem, ListItemText, Stack } from "@mui/material";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import React, { ReactNode } from "react";

export interface CardInfoProps {
  /** Generic bottom content _(will be displayed after `items`)_ */
  bottomContent?: ReactNode;
  /** Main card icon */
  icon?: ReactNode;
  /** Structured list of `CardInfoItem` */
  items?: CardInfoItem[];
  /** Main card title */
  title?: string;
  /** Generic top content _(will be displayed before `items`)_ */
  topContent?: ReactNode;
}

export interface CardInfoItemsProps {
  items: CardInfoItem[];
}

export interface CardInfoItem {
  /** footer text _(will be displayed on bottom)_ */
  footerText?: string;
  /** Item label */
  label: string;
  /** Item value */
  value: ReactNode | string;
}

const DATA_TEST_ID_PREFIX = "io-ipatente-card-info";
const TEXT_SECONDARY_COLOR = "text.secondary";

/** Basic info card component */
export const CardInfo = ({
  bottomContent,
  icon,
  items,
  title,
  topContent,
}: CardInfoProps) => (
  <Card
    data-testid={DATA_TEST_ID_PREFIX}
    sx={{ backgroundColor: "background.default", width: "100%" }}
  >
    <CardContent sx={{ padding: 2 }}>
      <Stack
        alignItems="center"
        direction="row"
        fontSize={24} // useful for inherited icon fontSize
        justifyContent="flex-start"
        spacing={1}
      >
        {icon}
        <Typography
          data-testid={`${DATA_TEST_ID_PREFIX}-title`}
          fontSize={20}
          fontWeight={600}
        >
          {title}
        </Typography>
      </Stack>
      <Stack
        alignItems="flex-start"
        direction="column"
        justifyContent="flex-start"
        marginTop={1.5}
        spacing={1}
      >
        {topContent}
        <CardInfo.Items items={items ?? []} />
        {bottomContent}
      </Stack>
    </CardContent>
  </Card>
);

/** Card Info items internal component */
const Items = ({ items }: CardInfoItemsProps) => (
  <List
    data-testid={`${DATA_TEST_ID_PREFIX}-items`}
    sx={{ padding: 0, width: "100%" }}
  >
    {items.map((item, index, items) => (
      <React.Fragment key={`${DATA_TEST_ID_PREFIX}-item-${index}`}>
        <ListItem disablePadding>
          <ListItemText
            disableTypography
            primary={
              <Typography
                color={TEXT_SECONDARY_COLOR}
                data-testid={`${DATA_TEST_ID_PREFIX}-item-${index}-label`}
                fontSize={18}
                fontWeight={600}
              >
                {item.label}
              </Typography>
            }
            secondary={
              React.isValidElement(item.value) ? (
                item.value
              ) : (
                <Typography
                  data-testid={`${DATA_TEST_ID_PREFIX}-item-${index}-value`}
                  fontSize={20}
                  fontWeight={600}
                >
                  {item.value}
                </Typography>
              )
            }
          />
        </ListItem>
        {item.footerText && (
          <Typography
            color={TEXT_SECONDARY_COLOR}
            dangerouslySetInnerHTML={{
              __html: item.footerText,
            }}
            data-testid={`${DATA_TEST_ID_PREFIX}-item-${index}-footer-text`}
            fontSize={12}
            fontWeight={600}
            marginTop={1.5}
          ></Typography>
        )}
        {index < items.length - 1 ? <Divider sx={{ marginY: 1.5 }} /> : null}
      </React.Fragment>
    ))}
  </List>
);

// namespaced component
CardInfo.Items = Items;
