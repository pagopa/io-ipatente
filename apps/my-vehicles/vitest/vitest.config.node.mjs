import { configDefaults, defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // FE
        "src/hooks/*",
        "src/layouts/*",
      ],
      extension: configDefaults.coverage.extension
        ?.toString()
        .replace(",.tsx", "")
        .split(","),
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
