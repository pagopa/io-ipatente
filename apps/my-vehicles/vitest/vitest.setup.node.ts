/**
 * MSW configuration for Vitest (used as setupFile in vitest.config.node.ts)
 */
import { loadEnvConfig } from "@next/env";
import { afterAll, afterEach, beforeAll } from "vitest";

import { mswServerTest } from "../mocks/msw-server";

let mswServer: ReturnType<typeof mswServerTest>;
// Start server before all tests
beforeAll(() => {
  loadEnvConfig(process.cwd());
  // this done to correctly load the environment variables on msw handlers
  mswServer = mswServerTest();
  mswServer.listen({ onUnhandledRequest: "warn" });
});

//  Close server after all tests
afterAll(() => mswServer.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers());
