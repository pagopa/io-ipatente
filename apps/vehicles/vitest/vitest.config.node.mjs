import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",
        "src/app/api/auth/mocks/**",
        // FE
        "src/common/*",
        "src/hooks/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.ts"],
      reporter: ["text", "html"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    env: loadEnv("test", process.cwd(), ""),
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
