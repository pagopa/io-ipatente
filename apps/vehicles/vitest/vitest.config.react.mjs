/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";
import { loadEnv } from "vite";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // BE
        "src/app/*",
        "src/lib/be/*",
      ],
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/frontend`,
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
