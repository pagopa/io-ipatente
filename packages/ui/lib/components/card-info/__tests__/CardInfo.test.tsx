import CheckCircle from "@mui/icons-material/CheckCircle";
import VerifiedUser from "@mui/icons-material/VerifiedUser";
import Chip from "@mui/material/Chip";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { CardInfo } from "../CardInfo";

const bottomContent = (
  <div data-testid="sample-bottom-content">bottom-content</div>
);
const topContent = <div data-testid="sample-top-content">top-content</div>;
const icon = <VerifiedUser fontSize="inherit" />;
const title = "sample-title";
const items = [
  { label: "label-1", value: "value-1" },
  {
    label: "label-2",
    value: (
      <Chip
        color="success"
        icon={<CheckCircle />}
        label="custom-label"
        size="small"
      />
    ),
  },
  {
    footerText: "sample-footer-text",
    label: "label-3",
    value: "value-3",
  },
];

describe("Test CardInfo Component", () => {
  it("Should match the snapshot with all props", () => {
    const comp = render(
      <CardInfo
        bottomContent={bottomContent}
        icon={icon}
        items={items}
        title={title}
        topContent={topContent}
      />,
    );
    expect(comp).toMatchSnapshot();
  });
});
