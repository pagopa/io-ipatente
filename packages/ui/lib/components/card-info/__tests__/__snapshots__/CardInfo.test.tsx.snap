// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test CardInfo Component > Should match the snapshot with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-tjaq38-MuiPaper-root-MuiCard-root"
        data-testid="io-ipatente-card-info"
      >
        <div
          class="MuiCardContent-root css-2awovp-MuiCardContent-root"
        >
          <div
            class="MuiStack-root css-ke8ruz-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="VerifiedUserIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm-2 16-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
              data-testid="io-ipatente-card-info-title"
            >
              sample-title
            </p>
          </div>
          <div
            class="MuiStack-root css-tm4a66-MuiStack-root"
          >
            <div
              data-testid="sample-top-content"
            >
              top-content
            </div>
            <ul
              class="MuiList-root MuiList-padding css-eui95f-MuiList-root"
              data-testid="io-ipatente-card-info-items"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                    data-testid="io-ipatente-card-info-item-0-label"
                  >
                    label-1
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
                    data-testid="io-ipatente-card-info-item-0-value"
                  >
                    value-1
                  </p>
                </div>
              </li>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-a0txjx-MuiDivider-root"
              />
              <li
                class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                    data-testid="io-ipatente-card-info-item-1-label"
                  >
                    label-2
                  </p>
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-txuvm7-MuiChip-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSuccess css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckCircleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                      />
                    </svg>
                    <span
                      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                    >
                      custom-label
                    </span>
                  </div>
                </div>
              </li>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-a0txjx-MuiDivider-root"
              />
              <li
                class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                    data-testid="io-ipatente-card-info-item-2-label"
                  >
                    label-3
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
                    data-testid="io-ipatente-card-info-item-2-value"
                  >
                    value-3
                  </p>
                </div>
              </li>
              <span
                class="MuiTypography-root MuiTypography-caption css-rb72w1-MuiTypography-root"
                data-testid="io-ipatente-card-info-item-2-footer-text"
              >
                sample-footer-text
              </span>
            </ul>
            <div
              data-testid="sample-bottom-content"
            >
              bottom-content
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-tjaq38-MuiPaper-root-MuiCard-root"
      data-testid="io-ipatente-card-info"
    >
      <div
        class="MuiCardContent-root css-2awovp-MuiCardContent-root"
      >
        <div
          class="MuiStack-root css-ke8ruz-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="VerifiedUserIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm-2 16-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
            data-testid="io-ipatente-card-info-title"
          >
            sample-title
          </p>
        </div>
        <div
          class="MuiStack-root css-tm4a66-MuiStack-root"
        >
          <div
            data-testid="sample-top-content"
          >
            top-content
          </div>
          <ul
            class="MuiList-root MuiList-padding css-eui95f-MuiList-root"
            data-testid="io-ipatente-card-info-items"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                  data-testid="io-ipatente-card-info-item-0-label"
                >
                  label-1
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
                  data-testid="io-ipatente-card-info-item-0-value"
                >
                  value-1
                </p>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-a0txjx-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                  data-testid="io-ipatente-card-info-item-1-label"
                >
                  label-2
                </p>
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-txuvm7-MuiChip-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSuccess css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                    />
                  </svg>
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    custom-label
                  </span>
                </div>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-a0txjx-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1yc420i-MuiTypography-root"
                  data-testid="io-ipatente-card-info-item-2-label"
                >
                  label-3
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-cjx14g-MuiTypography-root"
                  data-testid="io-ipatente-card-info-item-2-value"
                >
                  value-3
                </p>
              </div>
            </li>
            <span
              class="MuiTypography-root MuiTypography-caption css-rb72w1-MuiTypography-root"
              data-testid="io-ipatente-card-info-item-2-footer-text"
            >
              sample-footer-text
            </span>
          </ul>
          <div
            data-testid="sample-bottom-content"
          >
            bottom-content
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
